### ISSUE TITLE
Agent stops sending logs after upgrade to v3.4.15

### ISSUE THREAD

**Issue by mkowa42** [microsoft/ApplicationInsights-Java#3235]
### Expected behavior
Application logs collected by the agent are sent to Application Insights.

### Actual behavior
After upgrading the agent from 3.4.**14** to version 3.4.**15** the Application Insights is no longer showing any logs. Other types of events (e.g. requests, dependencies, exceptions, (custom) metrics) are still displayed. It seems to be that everything is sent to AI but no logs.

The application is showing warning and error messages. Refer to Logs below. These messages are not shown when using the previous agent version.

### System information
Please provide the following information:
 - SDK Version: 3.4.15
 - OS type and version: Linux container
 - Application Server type and version (if applicable): Quarkus 2.16.8, openjdk-17.0.8
 - Using spring-boot? NO
 - Additional relevant libraries (with version, if applicable):

### Logs

`APPLICATIONINSIGHTS_CONNECTION_STRING=InstrumentationKey=...;IngestionEndpoint=https://westeurope-2.in.applicationinsights.azure.com/;LiveEndpoint=https://westeurope.livediagnostics.monitor.azure.com/`

```
INFO exec -a "java" java -Dquarkus.http.host=0.0.0.0 -Djava.util.logging.manager=org.jboss.logmanager.LogManager -Djdk.tls.client.protocols=TLSv1.2 -XX:MaxRAMPercentage=85.0 -XX:+UseParallelGC -XX:MinHeapFreeRatio=10 -XX:MaxHeapFreeRatio=20 -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -XX:+ExitOnOutOfMemoryError -javaagent:/deployments/application-insights/applicationinsights-agent.jar -cp "." -jar /deployments/quarkus-run.jar 

2023-08-03 07:34:28.702Z INFO  c.m.applicationinsights.agent - Application Insights Java Agent 3.4.15 started successfully (PID 1, JVM running for 7.953 s)
2023-08-03 07:34:28.710Z INFO  c.m.applicationinsights.agent - Java version: 17.0.8, vendor: Red Hat, Inc., home: /usr/lib/jvm/java-17-openjdk-17.0.8.0.7-2.el8.x86_64
2023-08-03 07:34:30.913Z WARN  c.m.a.a.i.p.PerformanceMonitoringService - INITIALISING JFR PROFILING SUBSYSTEM THIS FEATURE IS IN BETA
2023-08-03 07:34:35.087Z WARN  c.a.m.o.e.i.p.TelemetryPipeline - Sending telemetry to the ingestion service: Received response code 400 (103: Field 'time' on type 'Envelope' is older than the allowed min date. Expected: now - 172800000ms) (future warnings will be aggregated and logged once every 5 minutes)
2023-08-03 07:34:34,939 INFO  [io.mic.cor.ins.pus.PushMeterRegistry] (main) ({}) publishing metrics for AzureMonitorMeterRegistry every 1m
2023-08-03 07:36:10.290Z ERROR c.azure.core.http.policy.RetryPolicy - {"az.sdk.message":"Retry attempts have been exhausted.","exception":"connection timed out: /169.254.169.254:80","tryCount":3}
2023-08-03 07:39:35.090Z WARN  c.a.m.o.e.i.p.TelemetryPipeline - In the last 5 minutes, the following operation has failed 3 times (out of 51): Sending telemetry to the ingestion service:
 * Received response code 206 (103: Field 'time' on type 'Envelope' is older than the allowed min date. Expected: now - 172800000ms) (3 times)
```

**Comment by mkowa42** [comment]
Hi @trask!

I turned on the self-diagnostics using DEBUG level.

In case of previous agent/sdk **version 3.4.14** it logs tons of lines like:

```
DEBUG c.m.a.a.i.exporter.AgentLogExporter - exporting log: com.microsoft.applicationinsights.agent.internal.init.TimestampingLogRecordProcessor ...`.
```

In case of current **version 3.4.15** it logs similar count of lines but slightly different content. Instead of `TimestampingLogRecordProcessor` it's a `SdkLogRecordData`. 

For example:

```
DEBUG c.m.a.a.i.exporter.AgentLogExporter - exporting log: SdkLogRecordData{resource=Resource{schemaUrl=https://opentelemetry.io/schemas/1.20.0, attributes={container.id="fa9ec41fe5c447dd0721b5ddcfc00dc7a711a9ba66ec426d94d6ee98bed84209", host.arch="amd64", host.name="fa9ec41fe5c4", os.description="Linux 5.10.102.1-microsoft-standard-WSL2", os.type="linux", process.command_args=[/usr/lib/jvm/java-17-openjdk-17.0.8.0.7-2.el8.x86_64/bin/java, -Dquarkus.http.host=0.0.0.0, -Djava.util.logging.manager=org.jboss.logmanager.LogManager, -Djavax.net.ssl.trustStore=/deployments/cacerts.jks, -Djdk.tls.client.protocols=TLSv1.2, -XX:MaxRAMPercentage=85.0, -XX:+UseParallelGC, -XX:MinHeapFreeRatio=10, -XX:MaxHeapFreeRatio=20, -XX:GCTimeRatio=4, -XX:AdaptiveSizePolicyWeight=90, -XX:+ExitOnOutOfMemoryError, -javaagent:/deployments/application-insights/applicationinsights-agent.jar, -cp, ., -jar, /deployments/quarkus-run.jar], process.executable.path="/usr/lib/jvm/java-17-openjdk-17.0.8.0.7-2.el8.x86_64/bin/java", process.pid=1, process.runtime.description="Red Hat, Inc. OpenJDK 64-Bit Server VM 17.0.8+7-LTS", process.runtime.name="OpenJDK Runtime Environment", process.runtime.version="17.0.8+7-LTS", service.name="xyz", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.28.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=org.jboss.logging, version=null, schemaUrl=null, attributes={}}, timestampEpochNanos=0, observedTimestampEpochNanos=1691414532812593000, spanContext=ImmutableSpanContext{traceId=00000000000000000000000000000000, spanId=0000000000000000, traceFlags=00, traceState=ArrayBasedTraceState{entries=[]}, remote=false, valid=false}, severity=DEBUG, severityText=DEBUG, body=StringBody{asString=Logging Provider: org.jboss.logging.JBossLogManagerProvider}, attributes={thread.id=1, thread.name="main"}, totalAttributeCount=2}
```

Any idea? Thanks!

**Comment by trask** [comment]
thanks! I was able to find the issue, and sent a fix upstream: https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/9159

this fix should make it into this month's release (3.4.16)

**Comment by mkowa42** [comment]
> thanks! I was able to find the issue, and sent a fix upstream: [open-telemetry/opentelemetry-java-instrumentation#9159](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/9159)
> 
> this fix should make it into this month's release (3.4.16)

Many thanks for your support! Looking forward to test again with next version.

**Comment by mtaghavi2005** [comment]
Facing the same issue, glad to see that it has been reported, and the fix will be released soon.

**Comment by trask** [comment]
this should be fixed now in 3.4.17
